Utils/github_workflow_scripts/run_secrets_detection_get_playbook_status.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import ast
+import sys
+import time
+import logging
+
+import demisto_client
+
+import urllib3
+from demisto_client.demisto_api.rest import ApiException
+from demisto_sdk.commands.common.constants import PB_Status
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+# ----- Constants ----- #
+DEFAULT_TIMEOUT = 60
+DEFAULT_INTERVAL = 20
+GOLD_SERVER_URL = "https://content-gold.paloaltonetworks.com"
+
+
+def get_playbook_state(client, inv_id):
+    # returns current investigation playbook state - 'inprogress'/'failed'/'completed'
+    try:
+        investigation_playbook_raw = demisto_client.generic_request_func(self=client, method='GET',
+                                                                         path='/inv-playbook/' + inv_id)
+        investigation_playbook = ast.literal_eval(investigation_playbook_raw[0])
+    except ApiException:
+        logging.exception('Failed to get investigation playbook state, error trying to communicate with demisto server')
+        return PB_Status.FAILED
+
+    try:
+        state = investigation_playbook['state']
+        return state
+    except:  # noqa: E722
+        return PB_Status.NOT_SUPPORTED_VERSION
+
+
+def wait_for_playbook_to_complete(investigation_id, client):
+    investigation_url = f'{GOLD_SERVER_URL}/#/WorkPlan/{investigation_id}'
+    logging.info(f'Investigation URL: {investigation_url}')
+
+    timeout = time.time() + DEFAULT_TIMEOUT
+
+    # wait for playbook to finish run
+    while True:
+        # give playbook time to run
+        time.sleep(1)
+        try:
+            playbook_state = get_playbook_state(client, investigation_id)
+        except demisto_client.demisto_api.rest.ApiException:
+            playbook_state = 'Pending'
+            client = demisto_client.configure(base_url=client.api_client.configuration.host,
+                                              api_key=client.api_client.configuration.api_key, verify_ssl=False)
+
+        if playbook_state == PB_Status.COMPLETED:
+            break
+
+        if playbook_state == PB_Status.FAILED:
+            print(f'Secrets playbook was failed as secrets were found. To investigate go to: {investigation_url} and'
+                  f' see SecretsDetection.FoundSecrets in the Context data.')
+            sys.exit(1)
+
+        if time.time() > timeout:
+            print(f'Secrets playbook timeout reached. To investigate go to: {investigation_url}')
+            sys.exit(1)
+
+    if playbook_state == PB_Status.COMPLETED:
+        print("Secrets playbook finished successfully, no secrets were found.")
```

Comment:
Let's move it to the if scope in 56 since you're already checking the state there.